src/storage/storage.cpp

```diff
@@ -52,7 +52,10 @@ aos::Error Storage::AddInstance(const aos::InstanceInfo& instance)
 
     auto storageInstance = ConvertInstanceInfo(instance);
 
-    return mInstanceDatabase.Add(*storageInstance);
+    return mInstanceDatabase.Add(
+        *storageInstance, [](const InstanceInfo& storedInstance, const InstanceInfo& addedInstance) {
+            return storedInstance == addedInstance || storedInstance.mInstanceIdent == addedInstance.mInstanceIdent;
```

Comment:
Looks not clear enough, doesn't "storedInstance == addedInstance" include "storedInstance.mInstanceIdent == addedInstance.mInstanceIdent" ?

Reply:
Removed compare the whole structure, left only compare PK. I can't use only compare the whole struct because it returns false if PK is the same but some fields changed